src/style-spec/function/index.js

```diff
@@ -53,19 +53,37 @@ type StylePropertySpecification = {
     value: 'number' | 'string' | 'boolean',
     length?: number,
     default?: Array<mixed>
+} | {
+    type: 'color',
+    default?: string
 };
 
 type StylePropertyValue = null | string | number | Array<string> | Array<number>;
 type FunctionParameters = DataDrivenPropertyValueSpecification<StylePropertyValue>
 
 function createFunction(parameters: FunctionParameters, propertySpec: StylePropertySpecification): StyleFunction {
-    let expr;
+    if (typeof parameters === 'string' && propertySpec.type === 'color') {
+        const color = parseColor(parameters);
+        return {
+            isFeatureConstant: true,
+            isZoomConstant: true,
+            evaluate() { return color; }
+        };
+    }
 
+    if (parameters === null || typeof parameters !== 'object' || Array.isArray(parameters)) {
```

Comment:
Is it ever possible, when a variable is not an object, but is an array?

Reply:
Not sure I understand the question. `typeof [] === 'object'`, right?

Reply:
Yes, I can't imagine, when `typeof x !== 'object' && Array.isArray(x)` is true